File: Backend/src/main/java/com/github/francisfire/anavis/models/ClosedPrenotation.java
package com.github.francisfire.anavis.models;

import java.util.Date;

import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Document(collection = "closedprenotation")
public class ClosedPrenotation extends Prenotation {

	private String reportId;

	public ClosedPrenotation(String id, String officeMail, String donorMail, Date hour, String reportId) {
		super(id, officeMail, donorMail, hour);
		this.reportId = reportId;
	}
}

File: Frontend/lib/models/closedprenotation.dart
import 'package:anavis/models/prenotation.dart';

class ClosedPrenotation extends Prenotation {
  String _reportId;
  ClosedPrenotation(String id, String officeMail, String donorMail, String hour,
      String reportId)
      : _reportId = reportId,
        super.withParams(id, officeMail, donorMail, hour);

  String getReportId() {
    return _reportId;
  }
}
